Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.cpp

```diff
@@ -266,11 +266,10 @@ GraphicsContext& RemoteImageBufferProxy::context() const
 void RemoteImageBufferProxy::putPixelBuffer(const PixelBuffer& pixelBuffer, const IntRect& srcRect, const IntPoint& destPoint, AlphaPremultiplication destFormat)
 {
     if (canMapBackingStore()) {
+        // Simulate a write so that pending reads migrate the data off of the mapped buffer.
+        context().fillRect({ });
         const_cast<RemoteImageBufferProxy&>(*this).flushDrawingContext();
         ImageBuffer::putPixelBuffer(pixelBuffer, srcRect, destPoint, destFormat);
-        // Simulate a write so that read caches are cleared.
-        // FIXME: This should not be done via the context draw, as that induces a flush.
-        context().fillRect({ });
```

Comment:
Does this change affect drawing this canvas into another canvas after putting the data? Can we test something like this?
```
    const ctxA = a.getContext('2d');
    const ctxB = b.getContext('2d');
    const ctxC = b.getContext('2d');
    ctxB.drawImage(a, 0, 0);
    ctxA.putImageData(imageData, 0, 0); // This should not affect canvas b.
    ctxC.drawImage(a, 0, 0); // This should get the latest ImageData in a.
```

Reply:
Since in the test there's no complex context state and the fillRect is 0,0,0,0, it's unlikely it would test the scenario. So in order to keep the repro case minimal, I'll omit this. I can try to find time to think if that and other cases like it is something we should test separately.